### PR TITLE
DC website compare tool navigation support

### PR DESCRIPTION
**Pull request by shifucun**
With this, when clicking in one side of the iframe, the other iframe can be updated correspondingly based on the url path.

### CODE REVIEW

File: server/templates/base.html [hunk 1]
```diff
@@ -73,6 +73,21 @@
     globalThis.isCustomDC = {{ config['CUSTOM']|int }};
     globalThis.STAT_VAR_HIERARCHY_CONFIG = {{ config['STAT_VAR_HIERARCHY_CONFIG'] | tojson }};
   </script>
+  <script>
+    // Post the iframe's URL to the parent
```
Comment (dwnoble) [inline server/templates/base.html:77]: What does this block of code do?

File: server/templates/base.html [hunk 2]
```diff
@@ -73,6 +73,21 @@
...
+    window.addEventListener('message', function(event) {
+      // MUST verify the origin to be autopush DC
+      if (event.origin === 'https://autopush.datacommons.org') {
```
Comment (dwnoble) [inline server/templates/base.html:80]: What's the check for autopush.datacommons.org for?

File: static/js/apps/diff/app.tsx [hunk 3]
```diff
@@ -42,20 +42,35 @@ export function App(): JSX.Element {
   const [url, setUrl] = useState<AppUrl>(INIT_URL);
   const [inputUrl, setInputUrl] = useState<AppUrl>(INIT_URL);
 
+  useEffect(() => {
```
Comment (dwnoble) [inline static/js/apps/diff/app.tsx:45]: I'm not sure the event passing is necessary here. If the goal is to keep the two iframes in sync, could you just set a local react state variable in this component from the iframe's onLoad and have the the relevant iframe re-render based on that state?

File: server/templates/base.html (hunk 1 above)
Comment (shifucun) [reply server/templates/base.html:77]: This block is to pass the iframe site url back to the parent

File: server/templates/base.html (hunk 2 above)
Comment (shifucun) [reply server/templates/base.html:80]: basically only allow autopush to host this comparator app

File: static/js/apps/diff/app.tsx (hunk 3 above)
Comment (shifucun) [reply static/js/apps/diff/app.tsx:45]: I tried that but it's not possible due to cross origin check. Googled around and this seems to be a popular way of doing it.

File: static/js/apps/diff/app.tsx [hunk 4]
```diff
@@ -42,20 +42,35 @@ export function App(): JSX.Element {
...
+    const handleMessage = (event) => {
+      if (event.data.type === "URLResponse") {
+        const iframeURL = event.data.url;
+        const { domain, path } = splitUrl(iframeURL);
+        console.log(domain, path);
```
Comment (chejennifer) [inline static/js/apps/diff/app.tsx:50]: remove logging?

**Review by chejennifer — APPROVED** [review]
this is cool!

File: static/js/apps/diff/app.tsx (hunk 4 above)
Comment (shifucun) [reply static/js/apps/diff/app.tsx:50]: Done

File: server/templates/base.html (hunk 1 above)
Comment (dwnoble) [reply server/templates/base.html:77]: is there  a way to extract this out of base.html to a ts/tsx file? if not, can you add some comments giving context as to what the intent is of the code (e.g., this supports the /dev/diff tool so navigating between the two sides i synchronized)

Comment (shifucun) [reply server/templates/base.html:77]: Done

**Review by dwnoble — APPROVED** [review]

**Comment by shifucun** [comment]
thanks for review!

**Pull request merged**